### PR TITLE
feat: add checkbox column helper

### LINKED ISSUE #69: Checkbox column
It would be great to add support for creating a special kind of column that would contain checkboxes as the value, and also a checkbox in the header that would select / deselect all the checkboxes in that specific column

### PR DESCRIPTION
**Pull request by flang**
Given that Vaadin's Grid doesn't provide a way to know when its dataprovider changes, if Select All checkbox is visible, and in order to keep it in sync with grid data, user must explicitly refresh the CheckboxColumn invoking refresh() method.

Close #69

### CODE REVIEW

File: src/main/java/com/flowingcode/vaadin/addons/gridhelpers/CheckboxColumnGridHelper.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+/*
+ * This file incorporates work licensed under the Apache License, Version 2.0 from Vaadin Cookbook
```
Comment (javier-godoy) [inline src/main/java/com/flowingcode/vaadin/addons/gridhelpers/CheckboxColumnGridHelper.java:22]: Which recipe is this code based on?

**Review by javier-godoy — CHANGES_REQUESTED** [review]

File: src/main/java/com/flowingcode/vaadin/addons/gridhelpers/CheckboxColumnGridHelper.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (flang) [reply src/main/java/com/flowingcode/vaadin/addons/gridhelpers/CheckboxColumnGridHelper.java:22]: definitely not, gonna remove it

File: src/test/java/com/flowingcode/vaadin/addons/gridhelpers/AllFeaturesDemo.java [hunk 2]
```diff
@@ -92,8 +98,8 @@ public AllFeaturesDemo() {
         .setHidable(true);
     grid.getColumns().forEach(c -> c.setAutoWidth(true));
 
-    grid.setItems(TestData.initializeData());
     grid.setSelectionMode(SelectionMode.MULTI);
+    grid.setItems(TestData.initializeData());
```
[the remark this answers is not in the document]
Comment (flang) [reply src/test/java/com/flowingcode/vaadin/addons/gridhelpers/AllFeaturesDemo.java]: fixed

File: src/test/java/com/flowingcode/vaadin/addons/gridhelpers/AllFeaturesDemo.java [hunk 3]
```diff
@@ -178,12 +188,12 @@ public AllFeaturesDemo() {
     binder.forField(heightByRowsField).bind(this::getHeightByRows, this::setHeightByRows);
     updateHeightByRowsField(grid, heightByRowsField);
     heightByRowsField.setStepButtonsVisible(true);
-
+
```
[the remark this answers is not in the document]
Comment (flang) [reply src/test/java/com/flowingcode/vaadin/addons/gridhelpers/AllFeaturesDemo.java]: fixed

File: src/main/resources/META-INF/frontend/fcGridHelper/vaadin-checkbox.css [hunk 4]
```diff
@@ -0,0 +1,59 @@
+:host([theme~='checkbox-top']) .vaadin-checkbox-container [part="checkbox"] {
```
[the remark this answers is not in the document]
Comment (flang) [reply src/main/resources/META-INF/frontend/fcGridHelper/vaadin-checkbox.css]: fixed

File: src/main/java/com/flowingcode/vaadin/addons/gridhelpers/CheckboxColumnGridHelper.java (hunk 1 above)
Comment (flang) [reply src/main/java/com/flowingcode/vaadin/addons/gridhelpers/CheckboxColumnGridHelper.java]: fixed

**Review by javier-godoy — APPROVED** [review]

**Pull request merged**